Sources/Commands/SwiftTestTool.swift

```diff
@@ -211,10 +225,15 @@ public struct SwiftTestTool: SwiftTool {
         return (mode ?? .run(nil), opts)
     }
 
-    private func test(path: AbsolutePath, xctestArg: String? = nil) throws -> Bool {
-        guard isValidTestPath(path) else {
-            throw TestError.testsExecutableNotFound
-        }
+    /// Executes the XCTest binary with given arguments.
+    ///
+    /// - Parameters:
+    ///     - path: Path to a valid XCTest binary.
+    ///     - xctestArg: Arguments to pass to the XCTest binary.
+    ///
+    /// - Returns: True if execution exited with return code 0.
+    private func test(path: AbsolutePath, xctestArg: String? = nil) -> Bool {
+        precondition(isValidTestPath(path))
```

Comment:
Should this really be a precondition?  Unless I'm missing something, whether or not `path` refers to a valid test is a runtime condition, so it should be reported through user errors rather than preconditions?